nupic/algorithms/KNNClassifier.py

```diff
@@ -830,9 +1026,11 @@ def computeSVD(self, numSVDSamples=None, finalize=True):
     else:
       self._a = self._Memory.toDense()[:self._numPatterns]
 
-    self._mean = numpy.mean(self._a, axis=0)
+    self._mean = numpy.mean(self._a, axis=0).astype('float64')
```

Comment:
I had to force this datatype because some times these variables were float64, another times were float32 because changes in `_Memory` variable (which alternate  in nupic.array and NearestNeighbor during a same iteration). These different datatypes for `_mean`, `_s`, and `_vt`, causes the assert test for compare two instances fails.